common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -0,0 +1,834 @@
+     * @return The new KeyPairGenerator object.
+     * @throws NoSuchProviderException  If the specified Provider is not registered in the security
+     *                                  provider list.
+     * @throws NoSuchAlgorithmException If a KeyPairGeneratorSpi implementation for the specified
+     *                                  algorithm is not available for the specified Provider.
+     */
+    private static KeyPairGenerator getKeyPairGenerator(@NonNull final String alg,
+                                                        @NonNull final String provider)
+            throws NoSuchProviderException, NoSuchAlgorithmException {
+        return KeyPairGenerator.getInstance(alg, provider);
+    }
+
+    /**
+     * Initialize the provided {@link KeyPairGenerator}.
+     *
+     * @param ctx              The current application Context.
+     * @param keyPairGenerator The KeyPairGenerator to initialize.
+     * @param keySize          The RSA keysize.
+     * @param useStrongbox     True if StrongBox should be used, false otherwise. Please note that
+     *                         StrongBox may not be supported on all devices.
+     * @throws InvalidAlgorithmParameterException
+     */
+    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
+    private static void initialize(@NonNull final Context ctx,
+                                   @NonNull final KeyPairGenerator keyPairGenerator,
+                                   final int keySize,
+                                   final boolean useStrongbox) throws InvalidAlgorithmParameterException {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            initializePre23(ctx, keyPairGenerator, keySize);
+        } else {
+            initialize23(keyPairGenerator, keySize, useStrongbox);
+        }
+    }
+
+    @SuppressLint("InlinedApi")
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private static void initialize23(@NonNull final KeyPairGenerator keyPairGenerator,
+                                     final int keySize,
+                                     final boolean useStrongbox) throws InvalidAlgorithmParameterException {
+        KeyGenParameterSpec.Builder builder = new KeyGenParameterSpec.Builder(
+                KEYSTORE_ENTRY_ALIAS,
+                KeyProperties.PURPOSE_SIGN
+                        | KeyProperties.PURPOSE_VERIFY
+                        | KeyProperties.PURPOSE_ENCRYPT
+                        | KeyProperties.PURPOSE_DECRYPT
+        )
+                .setKeySize(keySize)
+                .setSignaturePaddings(KeyProperties.SIGNATURE_PADDING_RSA_PKCS1)
+                .setDigests(KeyProperties.DIGEST_SHA256);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && useStrongbox) {
+            Logger.verbose(
+                    TAG,
+                    "Attempting to apply StrongBox isolation."
+            );
+            builder = applyHardwareIsolation(builder);
```

Comment:
nit: builder.setIsStrongBoxBacked(true);

Reply:
(I personally think that one line function is sort of overkill, but I'm fine with either. Your call.)

Reply:
This one I am going to leave as-is as this _could_ be a future extensibility point (for instance, if custom Samsung, Huawei, or other APIs exist)